pilot/pkg/xds/debug.go

```diff
@@ -770,16 +758,17 @@ func (s *DiscoveryServer) configDump(conn *Connection, includeEds bool) (*admin.
 		return nil, err
 	}
 
-	listeners, err := generate(v3.ListenerType)
-	if err != nil {
-		return nil, err
-	}
 	dynamicActiveListeners := make([]*admin.ListenersConfigDump_DynamicListener, 0)
-	for _, cs := range listeners {
+	for _, listener := range dump[v3.ListenerType] {
+		config := &listenerv3.Listener{}
```

Comment:
The generator already returns a `Resource` object that has the name. IMO we should use that instead of unmarshalling it again. This would be done by making getConfigDumpByResourceType return Resource instead of Any (and we can convert to Any as needed)

Reply:
there were several places where secrets needed to be redacted, so the underlying Resource had to be Unmarshalled in order to modify the fields.  So the either the underlying Resource had to be overwritten with the redacted object, or the additional logic needed to be added for the listener logic.  I switched to your suggestion, PTAL